diff --git a/node_modules/@react-navigation/native/lib/module/useLinking.js b/node_modules/@react-navigation/native/lib/module/useLinking.js
index 11d1f4a..8d4f7e6 100644
--- a/node_modules/@react-navigation/native/lib/module/useLinking.js
+++ b/node_modules/@react-navigation/native/lib/module/useLinking.js
@@ -54,9 +54,12 @@ const createMemoryHistory = () => {
         if (item.path === path) {
           return i;
         }
-      }
+	  }
+		
 
-      return -1;
+	  // @listiani13 add this change - Instead of relying to the loop above, find matching path from the whole list of routes
+	  const index = items.findIndex((val) => (val ? val.path === path : false));
+      return Math.max(index, 0);
     },
 
     push(_ref2) {
@@ -528,7 +531,7 @@ export default function useLinking(ref, _ref4) {
               // We couldn't find an existing entry to go back to, so we'll go back by the delta
               // This won't be correct if multiple routes were pushed in one go before
               // Usually this shouldn't happen and this is a fallback for that
-              await history.go(historyDelta);
+              await history.go(Math.max(historyDelta, 0));
             } // Store the updated state as well as fix the path if incorrect
 
 
